src/mon/OSDMonitor.cc

```diff
@@ -7902,31 +7902,75 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
   return 0;
 }
 
-int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, ostream *ss)
```

Comment:
Well, the commit description is misleading – it's not a squash of *all* the commits in the src PR; for instance: the `get_num_acting_pgs()` change is missed.

Instead of a squashing & rewriting, it might be desirable to judge the backportability of dependencies and restrict the differences to just C++20.

Reply:
I did squash all the commits of this PR. The removed methods in `get_num_acting_pgs()` do not exist in Q. Therefore, it resulted in an empty commit.

The difference is that the changes were not reverted in main - they were fixed in https://github.com/ceph/ceph/pull/49507.
However, in Q, the changes were reverted first and only then fixed (in this PR).

Reply:
My understanding of events is following:

1. there was a bug fix that introduced a regression to both `main` and `quincy`.
2. In `main` the regression has been fixed by a follow-up while in `quincy` the PR got simply reverted. 
3. Therefore the `check_pg_num()` simplification substantially differs between `main` and `quincy`.

It looks that the quincy version has unsatisfied dependencies (the very original, reverted PR + the regression fix). If so, I would bring those commits again – it still might be clearer than a custom patch.

Reply:
cherry-picked the reverted commits as well, Thanks for this suggestion!